From 12c9d23e40db9717356ddd1de111cb6c49918600 Mon Sep 17 00:00:00 2001
From: Jerry Huang <Chang-Ming.Huang@freescale.com>
Date: Tue, 17 Apr 2012 09:38:55 +0800
Subject: [PATCH 126/162] powerpc/mpc85xx: p1022ds support the MTD for NOR and
 NAND flash

The compatilbe 'simple-bus' is removed from the latest DTS for NAND and
NOR flash partition, so we must add the new compatilbe support for p1022ds,
otherwise, the kernel can't parse the partition of NOR and NAND flash.

Signed-off-by: Jerry Huang <Chang-Ming.Huang@freescale.com>
[Kevin: Original patch taken from FSL
QorIQ-SDK-V1.3-SOURCE-20121114-yocto.iso tarball. Create a function
p1022_ds_publish_elbc_device to publish the nor and nand device
since all the other common devices are published via
mpc85xx_common_publish_devices in 3.4 kernel.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/platforms/85xx/p1022_ds.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/powerpc/platforms/85xx/p1022_ds.c b/arch/powerpc/platforms/85xx/p1022_ds.c
index 29b89f1..99e39a7 100644
--- a/arch/powerpc/platforms/85xx/p1022_ds.c
+++ b/arch/powerpc/platforms/85xx/p1022_ds.c
@@ -605,6 +605,17 @@ static void __init p1022_ds_setup_arch(void)
 
 machine_device_initcall(p1022_ds, mpc85xx_common_publish_devices);
 
+static struct of_device_id __initdata p1022_elbc_ids[] = {
+	{ .compatible = "fsl,p1022-elbc", },
+	{},
+}
+
+static int __init p1022_ds_publish_elbc_device(void)
+{
+	return of_platform_bus_probe(NULL, p1022_elbc_ids, NULL);
+}
+machine_arch_initcall(p1022_ds, p1022_ds_publish_elbc_device);
+
 static struct of_device_id __initdata p1022_pci_ids[] = {
 	{ .compatible = "fsl,p1022-pcie", },
 	{},
-- 
1.7.9.7

